src/server.c

```diff
@@ -7212,6 +7212,11 @@ int main(int argc, char **argv) {
         loadDataFromDisk();
         aofOpenIfNeededOnServerStart();
         aofDelHistoryFiles();
+        /* While loading data, we delay applying "appendonly" config change.
+         * If there was a config change while we were inside loadDataFromDisk()
+         * above, we'll apply it here. */
+        applyAppendOnlyConfig();
```

Comment:
does this mean that it has no effect on `debug LOADAOF`?
why no do this at the end of `loadAppendOnlyFiles()`? maybe we can add a variable to indicate the config changed.

Reply:
good point about debug commands. `debug reload` is another place to consider this. 

> why no do this at the end of loadAppendOnlyFiles()

I tried to avoid any logic change here: 

```
        aofLoadManifestFromDisk();
        loadDataFromDisk();
        aofOpenIfNeededOnServerStart();
        aofDelHistoryFiles();
```

I guess it is safe to do it in loadDataFromDisk().

Reply:
> why no do this at the end of loadAppendOnlyFiles()

I think I misunderstood you. So, config can change while loading RDB or AOF. We should call it after both of these.

I added `applyAppendOnlyConfig()` to `debug reload` and `debug loadaof`.